packages/gatsby-cli/src/reporter/prepare-stack-trace.ts

```diff
@@ -0,0 +1,194 @@
+
+  constructor(...args) {
+    super(...args)
+    this.codeFrame = ``
+  }
+}
+
+export async function prepareStackTrace(
+  error: Error,
+  source: string
+): Promise<ErrorWithCodeFrame> {
+  const newError = new ErrorWithCodeFrame(error)
+  const map = await new SourceMapConsumer(readFileSync(source, `utf8`))
+  const stack = stackTrace
+    .parse(newError)
+    .map(frame => wrapCallSite(map, frame))
+    .filter(
+      frame =>
+        `wasConverted` in frame &&
+        (!frame.getFileName() ||
+          !frame
+            .getFileName()
+            .match(/^webpack:\/+(lib\/)?(webpack\/|\.cache\/)/))
+    )
+
+  newError.codeFrame = getErrorSource(map, stack[0])
+  newError.stack =
+    `${newError.name}: ${newError.message}\n` +
+    stack.map(frame => `    at ${frame}`).join(`\n`)
+
+  return newError
+}
+
+function getErrorSource(
+  map: BasicSourceMapConsumer | IndexedSourceMapConsumer,
+  topFrame: stackTrace.StackFrame | IWrappedStackFrame
+): string {
+  const source = map.sourceContentFor(topFrame.getFileName(), true)
+  return source
+    ? codeFrameColumns(
+        source,
+        {
+          start: {
+            line: topFrame.getLineNumber(),
+            column: topFrame.getColumnNumber(),
+          },
+        },
+        {
+          highlightCode: true,
+        }
+      )
+    : ``
+}
+
+interface IWrappedStackFrame {
+  getFileName(): string
+  getLineNumber(): number
+  getColumnNumber(): number
+  getScriptNameOrSourceURL(): string
+  toString: typeof CallSiteToString
+  wasConverted: true
+}
+
+function wrapCallSite(
+  map: BasicSourceMapConsumer | IndexedSourceMapConsumer,
+  frame: stackTrace.StackFrame
+): IWrappedStackFrame | stackTrace.StackFrame {
+  const source = frame.getFileName()
+  if (!source) return frame
+
+  const position = getPosition({ map, frame })
+  if (!position.source) return frame
+
+  return {
+    getFileName: (): string => position.source || ``,
+    getLineNumber: (): number => position.line || 0,
+    getColumnNumber: (): number => (position.column || 0) + 1,
+    getScriptNameOrSourceURL: (): string => position.source || ``,
+    toString: CallSiteToString,
+    wasConverted: true,
+  }
+}
+
+function getPosition({
+  map,
+  frame,
+}: {
+  map: BasicSourceMapConsumer | IndexedSourceMapConsumer
+  frame: stackTrace.StackFrame
+}): NullableMappedPosition {
+  const line = frame.getLineNumber()
+  const column = frame.getColumnNumber()
+  return map.originalPositionFor({ line, column })
+}
+
+interface INativeStringThisMethods {
+  isNative(): boolean
+  isEval(): boolean
+  getEvalOrigin(): string
+  isConstructor(): boolean
+  isToplevel(): boolean
+}
+
+// This is copied almost verbatim from the V8 source code at
+// https://code.google.com/p/v8/source/browse/trunk/src/messages.js.
+function CallSiteToString(): string {
+  // @ts-ignore
+  const _this = this
+  const self = _this as
+    | INativeStringThisMethods
+    | IWrappedStackFrame
+    | stackTrace.StackFrame
```

Comment:
This is a bit ugly, but it's the only way I could get things to work.

Reply:
necessary evil :)